File: HashTable/HashTable.py
class HashTable:
    def __init__(self, size):
        self.size = size
        self.table = [[] for _ in range(size)]

    def _hash(self, key):
        return hash(key) % self.size

    def insert(self, key, value):
        index = self._hash(key)
        for pair in self.table[index]:
            if pair[0] == key:
                pair[1] = value
                return
        self.table[index].append([key, value])

    def search(self, key):
        index = self._hash(key)
        for pair in self.table[index]:
            if pair[0] == key:
                return pair[1]
        return None

    def delete(self, key):
        index = self._hash(key)
        for i, pair in enumerate(self.table[index]):
            if pair[0] == key:
                del self.table[index][i]
                return


# Пример использования
hash_table = HashTable(10)
hash_table.insert("key1", "value1")
hash_table.insert("key2", "value2")
hash_table.insert("key3", "value3")

print(hash_table.search("key2"))  # Вывод: "value2"
hash_table.delete("key2")
print(hash_table.search("key2"))  # Вывод: None (ключ удален)

File: HashTable/HashTable.cs
using System;
using System.Collections.Generic;

class HashTable
{
    private int size;
    private List<KeyValuePair<string, string>>[] table;

    public HashTable(int size)
    {
        this.size = size;
        table = new List<KeyValuePair<string, string>>[size];
        for (int i = 0; i < size; i++)
        {
            table[i] = new List<KeyValuePair<string, string>>();
        }
    }

    private int Hash(string key)
    {
        return Math.Abs(key.GetHashCode()) % size;
    }

    public void Insert(string key, string value)
    {
        int index = Hash(key);
        foreach (var pair in table[index])
        {
            if (pair.Key == key)
            {
                pair.Value = value;
                return;
            }
        }
        table[index].Add(new KeyValuePair<string, string>(key, value));
    }

    public string Search(string key)
    {
        int index = Hash(key);
        foreach (var pair in table[index])
        {
            if (pair.Key == key)
            {
                return pair.Value;
            }
        }
        return "None";
    }

    public void Delete(string key)
    {
        int index = Hash(key);
        table[index].RemoveAll(pair => pair.Key == key);
    }
}

class Program
{
    static void Main()
    {
        HashTable hashTable = new HashTable(10);
        hashTable.Insert("key1", "value1");
        hashTable.Insert("key2", "value2");
        hashTable.Insert("key3", "value3");

        Console.WriteLine("Search key2: " + hashTable.Search("key2"));  // Output: "value2"
        hashTable.Delete("key2");
        Console.WriteLine("Search key2 after removal: " + hashTable.Search("key2"));  // Output: "None" (key removed)
    }
}